### DISCUSSION TITLE
Coding

### DISCUSSION THREAD

**Discussion by echointhewin** [catgiggle/OctoPrint-GpioControl#26] [General]
Do I need to code the GPIO in the RPI, or does this take care of it

**Comment by catgiggle** [comment]
In general case the answer is no. You don't have to configure anything outside GPIO Control. It automatically configure GPIO to be an output. But it can be, that there are some additional features enabled for the pin by default or manually by another script or program. In that case you need to make sure the pin is GPIO. Please note that there is difference between pin and GPIO. Pin is a physical connector and can be configured as SPI pin (eg, SPI_SCLK), I2C (eg I2C_SDA) or many others - including GPIO. Most of pins are GPIO's by default.
